### PR TITLE
Features

### PR DESCRIPTION
**Pull request by pepa65**
- Smarthome and smartend commands (go to beginning/end of line/screen/file).
- Paste at character position (not line-based)

### CODE REVIEW

File: commands.c [hunk 1]
```diff
@@ -69,6 +69,18 @@ static void cmd_bol(const char *pf, char **args)
 	move_bol();
 }
 
+static void cmd_smarthome(const char *pf, char **args)
```
Comment (craigbarnes) [inline commands.c:72]: You're using 2 and 3 space indents here, but tabs everywhere else.

Comment (pepa65) [reply commands.c:72]: Aaaw... Select & paste without fixing... Thanks for alerting me!
(Do I now squash/rebase this too..?)

**Comment by pepa65** [comment]
I think I managed to squash that corrective commit.

**Comment by craigbarnes** [comment]
Feel free to submit these to https://github.com/dte-editor/dte (preferably as 2 separate pull requests).

**Pull request closed without merging**